src/lib/metrics.py

```diff
@@ -200,11 +210,66 @@ def __init__(self, **kwargs):
         monitoring_filter = VARIABLE_VAR_PATTERN.sub('', monitoring_filter)
         object.__setattr__(self, "monitoring_filter", monitoring_filter)
         object.__setattr__(self, "is_enabled",  kwargs.get("is_enabled", True))
+        object.__setattr__(self, "extension_name",  kwargs.get("extension_name", "Unknown Extension"))
 
     def __hash__(self):
         return hash((self.name, self.technology_name, self.feature_set, self.monitoring_filter))
 
 
+class AutodiscoveryGCPService(GCPService):
```

Comment:
Think if all the internal relations are really necessary or could be avoided. Here, this has many Dicts. We could come back later to this, maybe there's something to do to reduce it, maybe not. Just to double check